packages/extension/src/lib/extensionScripts.ts

```diff
@@ -1,29 +1,37 @@
-import 'content-scripts-register-polyfill';
-import { browser, ContentScripts } from 'webextension-polyfill-ts';
+import browser from 'webextension-polyfill';
 import {
   CreateRequestContentScripts,
   contentScriptKey,
 } from '@dailydotdev/shared/src/hooks';
 import { companionPermissionGrantedLink } from '@dailydotdev/shared/src/lib/constants';
 import { AnalyticsEvent } from '@dailydotdev/shared/src/lib/analytics';
 
-export const registerBrowserContentScripts =
-  (): Promise<ContentScripts.RegisteredContentScript> =>
-    browser.contentScripts.register({
+let hasInjectedScripts = false;
+
+export const registerBrowserContentScripts = async (): Promise<void> => {
+  if (hasInjectedScripts) {
+    return null;
+  }
+
+  await browser.scripting.registerContentScripts([
+    {
+      id: `companion-${Date.now()}`,
```

Comment:
Is there a better way to do this? Worried we will register a bunch of same content scripts and load them to user browser. I suspect this might be due to content script being registered in each tab/context our extension and companion are loaded.

Reply:
This looks to me as something we need to solve before production.

Reply:
good point @capJavert I'll do some digging into this one

Reply:
@DragosIorgulescu so I was just checking something by removing `Date.now()` above. To my surprise I could not get the error we had before. Maybe it was fixed in other way? So yeah, maybe just remove it then?

Reply:
Agree this cannot remain as is before publishing. It's too risky

Reply:
this should be properly handled now 👍